File: examples/governance_static_go/domain/created_agreement.go
package domain

import "time"

type CreatedAgreement struct {
	AggregateID string    `json:"aggregate_id"`
	AggregateId string `json:"aggregate_id"`
	ModelId string `json:"model_id"`
	DataSource string `json:"data_source"`
	Purpose string `json:"purpose"`
	ConsentType string `json:"consent_type"`
	EffectiveDate time.Time `json:"effective_date"`
	ExpirationDate time.Time `json:"expiration_date"`
	Restrictions []Restriction `json:"restrictions"`
	Status string `json:"status"`
	OccurredAt time.Time `json:"occurred_at"`
}

func (e CreatedAgreement) EventName() string { return "CreatedAgreement" }

func (e CreatedAgreement) GetOccurredAt() time.Time { return e.OccurredAt }

File: examples/governance/model_registry_domain/lib/model_registry_domain/data_usage_agreement/events/created_agreement.rb
module ModelRegistryDomain
  class DataUsageAgreement
    module Events
      class CreatedAgreement
        attr_reader :aggregate_id, :model_id, :data_source, :purpose, :consent_type, :effective_date, :expiration_date, :restrictions, :status, :occurred_at

        def initialize(aggregate_id: nil, model_id: nil, data_source: nil, purpose: nil, consent_type: nil, effective_date: nil, expiration_date: nil, restrictions: nil, status: nil)
          @aggregate_id = aggregate_id
          @model_id = model_id
          @data_source = data_source
          @purpose = purpose
          @consent_type = consent_type
          @effective_date = effective_date
          @expiration_date = expiration_date
          @restrictions = restrictions
          @status = status
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end
